client/src/utils/api.ts

```diff
@@ -0,0 +1,278 @@
+import downloadZip from "./downloadZip";
+
+const BASE_URL = 'http://localhost:8080';
+interface CustomResponse {
+	zip?: string;
+	error?: boolean;
+	errorData?: string;
+	errorTitle?: string;
+	message?: string;
+}
+type StatusProps = {
+	message: string;
+	isSuccessful: boolean;
+	isEnd: boolean;
+};
+
+const settings = {
+	notificationLoaded: '.loaded',
+	notificationFailed: '.loaded-fail',
+	statusName: {
+		functionCreated: 'Function Created',
+		zipCreated: 'Zip Created',
+		zipUploaded: 'Zip Uploaded',
+		lambdaCreated: 'Lambda Created',
+		rangeTimeAdded: 'Range time added',
+	},
+	endPointUrls: {
+		modifyFileUrl: '/api/modify-file',
+		createZipUrl: '/api/create-zip',
+		uploadZipUrl: '/api/uploadZip',
+		createLambdaUrl: '/api/create-lambda',
+		addEventBridgeUrl: '/api/add-eventbridge',
+		modifyFileLocalUrl: '/api/modify-file-local',
+		createCfZip: '/api/create-cf-zip',
+	},
+};
+export type FetcherOptions = {
+	queryString: string
+	variables?: FetcherVariables
+}
+
+export type FetcherVariables = { [key: string]: string | any | undefined }
+
+
+
+class Api {
+	baseUrl: string;
+	constructor(baseUrl: string) {
+		this.baseUrl = baseUrl
+	}
+
+
+	customFetch = async (bodyToSend: object, url: string): Promise<CustomResponse> => {
+		return await fetch(`${BASE_URL}${url}`, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			...(Object.keys(bodyToSend).length > 0
+				? { body: JSON.stringify(bodyToSend) }
+				: {}),
+		})
+			.then((res) => {
+				if (res.ok) {
+					return res.json();
+				} else {
+					throw res;
+				}
+			})
+			.catch((err) => {
```

Comment:
this is the wrong approach. Each time on your response you will expect 2 objects - your response and the error. It will be hard to type with TS.